ledger/artifactmanager/handler.go

```diff
@@ -511,3 +513,41 @@ func validateState(old record.State, new record.State) error {
 	}
 	return nil
 }
+
+func (h *MessageHandler) handleGetHistory(ctx context.Context, pulseNumber core.PulseNumber, genericMsg core.SignedMessage) (core.Reply, error) {
+	msg := genericMsg.Message().(*message.GetHistory)
+
+	idx, _, _, err := getObject(ctx, h.db, msg.Object.Record(), nil, true)
+	if err != nil {
+		return nil, err
+	}
+	history := []record.ObjectState{}
```

Comment:
`record` structures should not be transferred via network. It's an internal structure. Let's create a separate data structure and pass it around.